src/IO/Archives/LibArchiveReader.cpp

```diff
@@ -147,7 +193,16 @@ class LibArchiveReader::Handle
         file_info.reset();
     }
 
-    static struct archive * open(const String & path_to_archive)
+    static struct archive * openWithReader(StreamInfo * read_stream_)
+    {
+        auto * a = archive_read_new();
+        archive_read_support_filter_all(a);
+        archive_read_support_format_all(a);
+        archive_read_open(a, read_stream_, nullptr, StreamInfo::read, nullptr);
+        return a;
+    }
```

Comment:
You could consider returning `std::unique_ptr` with a `close` as a custom deleter.

Reply:
I think it is probably better to leave as is for now as all libarchive functions take a bare `struct archive *`. I made `Handle` more self contained to not expose any bare `struct archive *`.